### PR TITLE
Init action on insertion

### PR DESCRIPTION
**Pull request by Clazex**
This is mainly for compatibility with my upcoming pmfsm optimization mod.

### CODE REVIEW

File: Transcendence/PlayMakerExtensions.cs [hunk 1]
```diff
@@ -63,7 +64,9 @@ internal static void PrependAction(this FsmState s, Action a)
 
         internal static void ReplaceAction(this FsmState s, int i, Action a)
         {
-            s.Actions[i] = new FuncAction(a);
+            FuncAction action = new FuncAction(a);
```
Comment (dpinela) [inline Transcendence/PlayMakerExtensions.cs:67]: I'll fix this myself on merging, but I usually use `var` to declare most variables; keep that in mind for future contributions here.

**Comment by dpinela** [comment]
What does the Init method on actions normally do, and why is it necessary for your mod to work?

**Comment by Clazex** [comment]
`FsmStateAction.Init` only assigns references to the state, fsm, component and game object.

My optimization is to make `FsmStateAction.Init` only be called within `Awake`, so if one add actions after `Awake`, those won't get `Init`ed, thus the fsm util must init the action.

File: Transcendence/PlayMakerExtensions.cs (hunk 1 above)
Comment (Clazex) [reply Transcendence/PlayMakerExtensions.cs:67]: I'll keep that in mind.

**Pull request merged**